.github/workflows/packages.yml

```diff
@@ -230,6 +230,59 @@ jobs:
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
         if: github.ref == 'refs/heads/master' && always() && matrix.publish == 'true' && (steps.build.outcome == 'success' || steps.build.outcome == 'skipped') && steps.check.outcome == 'success' && steps.install.outcome == 'success'
+#check dockerfiles & build starts !!!!!!
+      - name: Check dockerfiles directory existence
+        id: check_directory
+        uses: andstor/file-existence-action@v2
+        with:
+          files: "dockerfiles"  #directory is a file
+      - name: check if dockerfies directory not empty 
+        id: NEdockerdir
+        run: |
+          if [ -z "$(ls -A  packages/${{ matrix.package }}/dockerfiles)" ]; then
+          echo "DIR_EMPTY=true" >> $GITHUB_ENV
+          else
+          echo "DIR_EMPTY=false" >> $GITHUB_ENV
+          fi
+      - name: Check if directory has changes
+        id: directory_changes_checker
+        uses: georgeneto/directory-changes-checker@v1
+        with:
+          repo-token: ${{secrets.GITHUB_TOKEN}}
+          directory: "packages/${{ matrix.package }}/dockerfiles"
+      - name: login to DockerHub   
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }} 
+        if: ${{steps.check_directory.outputs.files_exists}} == 'true' && ${{env.DIR_EMPTY}} == 'false' && steps.publish.outcome == 'success'
+      - name: Build and push package dockerimages
+        working-directory: packages/${{ matrix.package }}
+        run: |
+           current_version=$(jq .version package.json | sed -E 's/(^"|"$)//g')
+           name=$(jq .name package.json | sed -E 's/(^"|"$)//g')
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: datagrok/${name}:${current_version}
+        if: ${{steps.check_directory.outputs.files_exists}} == 'true' && ${{env.DIR_EMPTY}} == 'false' && steps.publish.outcome == 'success'
+
+    #   - name: get image version
+     #     run: |
+     #       echo "IMAGE_VER=`echo "$(jq .version package.json | sed -E -e 's/(^"|"$)//g')"`" >> $GITHUB_ENV
+     #       echo "PACKAGE_NAME=`echo "$(jq .name package.json | sed -E -e 's/(^"|"$)//g' -e 's,@datagrok/,,g')"`" >> $GITHUB_ENV
+      - name: build dockerfiles as artifact when npm push failed
+        uses: docker/build-push-action@v3
+        with:
+          push: false
+          tags: datagrok/${name}:${current_version}
+        if: ${{steps.directory_changes_checker.output.hasChanges}} == 'true' && && steps.publish.outcome == 'fail'
```

Comment:
it does not make sense. What if publish step was skipped?

Reply:
it will bild an image, then image uploads as artifact at next step